src/cursor/abstract_cursor.ts

```diff
@@ -903,7 +866,11 @@ class ReadableCursorStream extends Readable {
   }
 
   override _destroy(error: Error | null, callback: (error?: Error | null) => void): void {
-    this._cursor.close(err => process.nextTick(callback, err || error));
+    // FIXME: how is this possible?
+    const res = this._cursor.close();
+    if (res != null) {
+      res.finally(() => callback(error));
+    }
```

Comment:
Still debugging this ([evergreen error link](https://evergreen.mongodb.com/task_log_raw/mongo_node_driver_next_rhel80_large_gallium_test_5.0_replica_set_patch_056c86f6dee36f805376513c8322ac1ecab3c8e6_63cc20c7d1fe07422f90da68_23_01_21_17_28_40/0?type=T#L5791)). Any ideas?

Reply:
which evergreen run was this from?  I did an (admittedly brief) skim of the evergreen patch and didn't see it


I did see that serverless is particularly bad, but I'm not sure when that started

Reply:
from `rhel80_large_gallium_test_5.0_replica_set` in [this patch](https://spruce.mongodb.com/task/mongo_node_driver_next_rhel80_large_gallium_test_5.0_replica_set_patch_056c86f6dee36f805376513c8322ac1ecab3c8e6_63cc20c7d1fe07422f90da68_23_01_21_17_28_40/tests?execution=0&sortBy=STATUS&sortDir=ASC)

But it's not the patch I'm referring to, the AbstractCursor.close method is an `async function` and yet it returns `undefined` the second time it is invoked, I have to gate calling finally on a nullish check. 🤔 super strange

Reply:
It was `sinon` 🤦  proxies aren't obvious until you step-into the function call